File: Competitions/Romário's Trello/474BWorms.py
def binSearch(array, lo, hi, target):
    mid = (lo + hi) // 2
    if (lo >= hi):
        return(mid // 2)
    if (array[mid] >= target):
        return(binSearch(array, lo, mid, target))
    return(binSearch(array, mid + 1, hi, target))

numPiles = int(input())
piles = list(map(int, input().split()))
worms, prev = [], 0
for i, j in enumerate(piles):
    worms += [prev + 1]
    worms += [prev + j]
    prev += j
numQueries = int(input())
queries = list(map(int, input().split()))
for i in queries:
    print(binSearch(worms, 0, numPiles * 2, i) + 1)

File: Competitions/Romário's Trello/474BWorms.cpp
#include <bits/stdc++.h>

int binsearch(int array[], int lo, int hi, int target)
{
  int mid = (lo + hi) / 2;
  if (lo >= hi) return(array[hi] == target ? hi : lo);
  if (target < array[mid]) return(binsearch(array, lo, mid - 1, target));
  else return(binsearch(array, mid, hi, target));
}

int main()
{
  int n; scanf("%d",  &n);
  int worms[n], pile, at = 1;
  for (int i = 0; i < n; i ++)
  {
    scanf("%d", &pile);
    worms[i] = at;
    at += pile;
  }

  int queries, q; scanf("%d", &queries);
  while (queries -- > 0)
  {
    scanf("%d", &q);
    printf("%d\n", binsearch(worms, 0, at, q));
  }

  return(0);
}